pkg/grpc/api.go

```diff
@@ -1377,15 +1392,22 @@ func (a *api) SubscribeConfiguration(request *runtimev1pb.SubscribeConfiguration
 		apiServerLogger.Debug(err)
 		return err
 	}
-	if err := handler.serverStream.Send(&runtimev1pb.SubscribeConfigurationResponse{
+
+	err = handler.serverStream.Send(&runtimev1pb.SubscribeConfigurationResponse{
 		Id: subscribeID,
-	}); err != nil {
+	})
+	if err != nil {
 		apiServerLogger.Debug(err)
 		return err
 	}
 
 	stop := make(chan struct{})
 	a.CompStore.AddConfigurationSubscribe(subscribeID, stop)
+
+	// We have sent the first message, so signal that we're ready to send messages in the stream
+	handler.ready()
+
+	// Wait until the channel is stopped
 	<-stop
```

Comment:
Does this close in shutdown scenarios? It doesn't look like it does when we delete the configuration subscription at least.

Reply:
See #6560. I'd like to tackle that separately